test/utils.dart

```diff
@@ -27,129 +28,51 @@ Watcher _watcher;
 /// and is polling for changes. If you pass `false` for [waitForReady], it will
 /// not schedule this delay.
 ///
 /// If [path] is provided, watches a subdirectory in the sandbox with that name.
-Watcher createWatcher({String path, bool waitForReady}) {
+Watcher createWatcher({String path}) {
   if (path == null) {
-    path = _sandboxDir;
+    path = d.sandbox;
   } else {
-    path = p.join(_sandboxDir, path);
+    path = p.join(d.sandbox, path);
   }
 
-  var watcher = _watcherFactory(path);
-
-  // Wait until the scan is finished so that we don't miss changes to files
-  // that could occur before the scan completes.
-  if (waitForReady != false) {
-    schedule(() => watcher.ready, "wait for watcher to be ready");
-  }
-
-  return watcher;
+  return _watcherFactory(path);
 }
 
 /// The stream of events from the watcher started with [startWatcher].
-ScheduledStream<WatchEvent> _watcherEvents;
+StreamQueue<WatchEvent> _watcherEvents;
 
 /// Creates a new [Watcher] that watches a temporary file or directory and
 /// starts monitoring it for events.
 ///
 /// If [path] is provided, watches a path in the sandbox with that name.
-void startWatcher({String path}) {
-  // We want to wait until we're ready *after* we subscribe to the watcher's
-  // events.
-  _watcher = createWatcher(path: path, waitForReady: false);
-
-  // Schedule [_watcher.events.listen] so that the watcher doesn't start
-  // watching [path] before it exists. Expose [_watcherEvents] immediately so
-  // that it can be accessed synchronously after this.
-  _watcherEvents = new ScheduledStream(futureStream(
-      schedule(() {
-        currentSchedule.onComplete.schedule(() {
-          _watcher = null;
-          if (!_closePending) _watcherEvents.close();
-
-          // If there are already errors, don't add this to the output and make
-          // people think it might be the root cause.
-          if (currentSchedule.errors.isEmpty) {
-            _watcherEvents.expect(isDone);
-          }
-        }, "reset watcher");
-
-        return _watcher.events;
-      }, "create watcher"),
-      broadcast: true));
-
-  schedule(() => _watcher.ready, "wait for watcher to be ready");
-}
+Future<Null> startWatcher({String path}) async {
+  mockGetModificationTime((path) {
+    path = p.normalize(p.relative(path, from: d.sandbox));
 
-/// Whether an event to close [_watcherEvents] has been scheduled.
-bool _closePending = false;
+    // Make sure we got a path in the sandbox.
+    assert(p.isRelative(path) && !path.startsWith(".."));
 
-/// Schedule closing the watcher stream after the event queue has been pumped.
-///
-/// This is necessary when events are allowed to occur, but don't have to occur,
-/// at the end of a test. Otherwise, if they don't occur, the test will wait
-/// indefinitely because they might in the future and because the watcher is
-/// normally only closed after the test completes.
-void startClosingEventStream() {
```

Comment:
Why was this deleted?

Reply:
This smells funny to me... why don't we need to cancel the stream anymore?

Reply:
They weren't necessary when I removed the Backup cases. Now that these are present this function is now necessary.